setuptools/dist.py

```diff
@@ -438,9 +439,22 @@ def __init__(self, attrs=None):
                 value = default() if default else None
             setattr(self.metadata, option, value)
 
-        self.metadata.version = self._validate_version(self.metadata.version)
+        self.metadata.version = self._normalize_version(
+            self._validate_version(self.metadata.version))
         self._finalize_requires()
 
+    @staticmethod
+    def _normalize_version(version):
+        if isinstance(version, setuptools.sic) or version is None:
```

Comment:
What happens if `version=''` as opposed to `None` or is not a string?

Reply:
I would expect in those cases for the behavior to be unchanged with this PR.